OpenRA.Mods.Common/Activities/FindResources.cs

```diff
@@ -49,59 +49,63 @@ public FindResources(Actor self, CPos avoidCell)
 
 		public override Activity Tick(Actor self)
 		{
+			if (ChildActivity != null)
+			{
+				ChildActivity = ActivityUtils.RunActivity(self, ChildActivity);
+				return this;
+			}
+
 			if (IsCanceling)
 				return NextActivity;
 
 			if (harv.IsFull)
-			{
-				// HACK: DeliverResources is ignored if there are queued activities, so discard NextActivity
-				return ActivityUtils.SequenceActivities(self, new DeliverResources(self));
-			}
+				return NextActivity;
 
 			var closestHarvestablePosition = ClosestHarvestablePos(self);
 
 			// If no harvestable position could be found, either deliver the remaining resources
 			// or get out of the way and do not disturb.
 			if (!closestHarvestablePosition.HasValue)
 			{
-				if (!harv.IsEmpty)
-					return new DeliverResources(self);
-
 				harv.LastSearchFailed = true;
+				var lastproc = harv.LastLinkedProc ?? harv.LinkedProc;
+				if (lastproc != null && !lastproc.Disposed)
+				{
+					var deliveryLoc = lastproc.Location + lastproc.Trait<IAcceptResources>().DeliveryOffset;
+					if (self.Location == deliveryLoc && harv.IsEmpty)
+					{
+						// Get out of the way:
+						var unblockCell = deliveryLoc + harv.Info.UnblockCell;
+						var moveTo = mobile.NearestMoveableCell(unblockCell, 1, 5);
+						self.SetTargetLine(Target.FromCell(self.World, moveTo), Color.Green, false);
+						QueueChild(self, mobile.MoveTo(moveTo, 1), true);
+						return this;
+					}
+				}
 
-				var unblockCell = harv.LastHarvestedCell ?? (self.Location + harvInfo.UnblockCell);
```

Comment:
This is the line that parks the harvester in the last harvested ore patch which would allow it to ignore `SearchFromProcRadius` altogether in certain maps. This is incidentally also the line that causes #16205.